src/scancode/outdated.py

```diff
@@ -143,7 +143,7 @@ def check_scancode_version_locally(
     Return a message to display if outdated or None. Work offline, without a
     PyPI remote check.
     """
-    if is_outdated(release_date):
+    if is_outdated(release_date) and installed_version<fetch_latest_version():
```

Comment:
That's smart! We need still to make sure that this is NOT called when offline. 
This is breaking the expectation laid above https://github.com/nexB/scancode-toolkit/pull/2800/files#diff-ce07c0640f6e9f4595b760563228f57f721c3c142fecbdd5a33b6efcbb2b9e3cR143

Reply:
I also also believe the check is wrong at this place. IMO we should have something like a wrapping `check_scancode_version` function that calls `check_scancode_version_locally` when offline and `check_scancode_version_remotely` when online.

Reply:
Ok I'll make the changes

Reply:
Thanks!
I suggest we discuss a little in https://github.com/nexB/scancode-toolkit/issues/2796 to find the correct design.

Reply:
Yess i was thinking the same becasue i was unable to think any solution for fixing the problem locally